In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv("D:\Semester 4\Data Analysis\Research Dataset\GapMinder Codebook.csv")

In [3]:
df.head(15)

country incomeperperson alcconsumption armedforcesrate  \
0           Afghanistan                           0.03       0.5696534   
1               Albania     1914.996551           7.29       1.0247361   
2               Algeria     2231.993335           0.69        2.306817   
3               Andorra      21943.3399          10.17                   
4                Angola     1381.004268           5.57       1.4613288   
5   Antigua and Barbuda     11894.46407           8.17                   
6             Argentina     10749.41924           9.35        0.560987   
7               Armenia     1326.741757          13.66       2.6184384   
8                 Aruba                                                  
9             Australia     25249.98606          10.21       0.4862799   
10              Austria     26692.98411           12.4       0.8155798   
11           Azerbaijan     2344.896916          13.34       1.9767462   
12              Bahamas     19630.54055           8.65       0.5452863   
13              Bahrain     12505.21254           4.19       5.2311429   
14           Bangladesh     558.0628766           0.17       0.2878915   

   breastcancerper100th co2emissions femaleemployrate hivrate internetuserate  \
0                  26.8     75944000      25.60000038             3.654121623   
1                  57.4  223747333.3      42.09999847             44.98994696   
2                  23.5   2932108667      31.70000076     0.1     12.50007331   
3                                                                          81   
4                  23.1    248358000      69.40000153       2     9.999953883   
5                           16225000                              80.64545455   
6                  73.9   5872119000      45.90000153     0.5     36.00033495   
7                  51.6  51219666.67      34.20000076     0.1     44.00102458   
8                           35871000                              41.80088889   
9                  83.2  12970092667      54.59999847     0.1      75.8956538   
10                 70.5   4466084333      49.70000076     0.3     72.73157554   
11                 31.5  511107666.7      56.20000076     0.1     46.67970157   
12                 54.4    137555000      60.70000076     3.1     42.98458017   
13                 40.2  503994333.3      30.20000076             54.99280903   
14                 16.6    598774000      53.59999847    0.06      3.70000326   

   lifeexpectancy oilperperson polityscore relectricperperson suicideper100th  \
0          48.673                        0                          6.6843853   
1          76.918                        9        636.3413834     7.699329853   
2          73.131  0.420094525           2        590.5098143     4.848769665   
3                                                                 5.362178802   
4          51.093                       -2        172.9992274     14.55467701   
5                                                                   2.1618433   
6          75.901  0.635943801           8        768.4282997        7.765584   
7          74.241                        5        603.7630576     3.741587877   
8          75.246                                                               
9          81.907  1.913026109          10        2825.391095     8.470030125   
10         80.854  1.548790966          10        2068.123309        13.09437   
11         70.739   0.35917261          -7        921.5621108     1.380964637   
12          75.62                                                 3.374415874   
13         75.057                       -7        7314.355637     4.414990425   
14         68.944  0.032281466           5        68.11522858     14.53835678   

     employrate urbanrate  
0   55.70000076     24.04  
1   51.40000153     46.72  
2          50.5     65.22  
3                   88.92  
4   75.69999695      56.7  
5                   30.46  
6   58.40000153        92  
7   40.099998

In [4]:
print("Total Columns : ",len(df.columns))
print("Total Row : ",len(df))

Total Columns :  16
Total Row :  213


In [5]:
df.dtypes

country                 object
incomeperperson         object
alcconsumption          object
armedforcesrate         object
breastcancerper100th    object
co2emissions            object
femaleemployrate        object
hivrate                 object
internetuserate         object
lifeexpectancy          object
oilperperson            object
polityscore             object
relectricperperson      object
suicideper100th         object
employrate              object
urbanrate               object
dtype: object

In [6]:
df["incomeperperson"][0]

' '

In [7]:
df["armedforcesrate"][3]

' '

In [8]:
df["breastcancerper100th"][3]

' '

This means all the cells given in the excel(.csv file) as blank are type of String.

In [9]:
df_opp = df[df["oilperperson"]== " "]

print("Empty cells in oilperperson column is :",len(df_opp))
print("Usuable cells are only: ", len(df["oilperperson"])-len(df_opp))

Empty cells in oilperperson column is : 150
Usuable cells are only:  63


This is not a good column because out of 213 rows, 150 rows are empty.

In [10]:
df_repp = df[df["relectricperperson"]== " "]

print("Empty cells in relectricperperson column is :",len(df_repp))
print("Usuable cells are only: ", len(df["relectricperperson"])-len(df_repp))

Empty cells in relectricperperson column is : 77
Usuable cells are only:  136


This is a good column to consider, lets change all the empty cells to NAN in these two columns

In [11]:
'''df = df[df["oilperperson"] == " "].replace(" ", np.nan)
df = df[df["relectricperperson"] == " "].replace(" ", np.nan)'''

'df = df[df["oilperperson"] == " "].replace(" ", np.nan)\ndf = df[df["relectricperperson"] == " "].replace(" ", np.nan)'

In [19]:
df.drop(df[df["oilperperson"] == " "].index, inplace = True)
df.drop(df[df["relectricperperson"] == " "].index, inplace = True)

In [20]:
df

country incomeperperson alcconsumption armedforcesrate  \
2                 Algeria     2231.993335           0.69        2.306817   
6               Argentina     10749.41924           9.35        0.560987   
9               Australia     25249.98606          10.21       0.4862799   
10                Austria     26692.98411           12.4       0.8155798   
11             Azerbaijan     2344.896916          13.34       1.9767462   
..                    ...             ...            ...             ...   
201  United Arab Emirates     21087.39412           0.52       1.8124763   
202        United Kingdom     28033.48928          13.24       0.5080181   
203         United States     37491.17952            9.7       0.9723782   
205            Uzbekistan     952.8272608           3.61       0.7096295   
207             Venezuela     5528.363114            7.6       0.9040245   

    breastcancerper100th co2emissions femaleemployrate hivrate  \
2                   23.5   2932108667      31.70000076     0.1   
6                   73.9   5872119000      45.90000153     0.5   
9                   83.2  12970092667      54.59999847     0.1   
10                  70.5   4466084333      49.70000076     0.3   
11                  31.5  511107666.7      56.20000076     0.1   
..                   ...          ...              ...     ...   
201                 24.1   2315698000      37.29999924           
202                 87.2  72524250333      53.09999847     0.2   
203                101.1     3.34E+11               56     0.6   
205                 17.3   1718339333      52.59999847     0.1   
207                 34.3   5418886000      45.79999924           

    internetuserate lifeexpectancy oilperperson polityscore  \
2       12.50007331         73.131  0.420094525           2   
6       36.00033495         75.901  0.635943801           8   
9        75.8956538         81.907  1.913026109          10   
10      72.73157554         80.854  1.548790966          10   
11      46.67970157         70.739   0.35917261          -7   
..              ...            ...          ...         ...   
201     77.99678115         76.546  4.299964455          -8   
202     84.73170475          80.17  1.188028094          10   
203       74.247572         78.531  2.738543683          10   
205     19.44502055         68.287  0.182184525          -9   
207     35.85043696         74.402  1.214636773          -3   

    relectricperperson suicideper100th   employrate urbanrate  
2          590.5098143     4.848769665         50.5     65.22  
6          768.4282997        7.765584  58.40000153        92  
9          2825.391095     8.470030125         61.5     88.74  
10         2068.123309        13.09437  57.09999847     67.16  
11         921.5621108     1.380964637  60.90000153     51.92  
..                 ...             ...          ...       ...  
201        11154.75503     1.392950654  75.19999695     77.88  
202        1933.945615        6.014659  59.29999924     89.94  
203        4542.848695        9.927033  62.29999924      81.7  
205        261.4339532         5.21372         57.5     36.82  
207        969.0043388         4.11962  59.90000153     93.32  

[63 rows x 16 columns]

We are going to find data about only 63 countries

In [24]:
df["oilpersperson"] = df["oilperperson"].astype(float)
df["relectricperperson"] = df["relectricperperson"].astype(float)
df.dtypes

country                  object
incomeperperson          object
alcconsumption           object
armedforcesrate          object
breastcancerper100th     object
co2emissions             object
femaleemployrate         object
hivrate                  object
internetuserate          object
lifeexpectancy           object
oilperperson             object
polityscore              object
relectricperperson      float64
suicideper100th          object
employrate               object
urbanrate                object
oilpersperson           float64
dtype: object

In [30]:
oppcount = df["oilperperson"].value_counts(sort = False).to_frame()
oppcount

oilperperson
0.420094525             1
0.635943801             1
1.913026109             1
1.548790966             1
0.35917261              1
...                   ...
4.299964455             1
1.188028094             1
2.738543683             1
0.182184525             1
1.214636773             1

[63 rows x 1 columns]

In [31]:
relectriccount = df["relectricperperson"].value_counts(sort = False).to_frame()
relectriccount

relectricperperson
590.509814                     1
768.428300                     1
2825.391095                    1
2068.123309                    1
921.562111                     1
...                          ...
11154.755030                   1
1933.945615                    1
4542.848695                    1
261.433953                     1
969.004339                     1

[63 rows x 1 columns]

This informs us that all the data in column of oilperperson and relectricperperson is Unique.

## Research Question : Is there any correlation between Oil-per-person and r-electric-per-person?